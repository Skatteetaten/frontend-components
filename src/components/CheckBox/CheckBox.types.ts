import { ICheckboxProps } from '@fluentui/react';

export interface CheckBoxProps extends ICheckboxProps {
  /** Angir teknisk navn p√• elementet */
  name?: string;
  /** Om checkboxen skal markeres med feiltilstand */
  error?: boolean;
  /** @ignore */
  indeterminate?: ICheckboxProps['indeterminate'];
  /** @ignore */
  defaultIndeterminate?: ICheckboxProps['indeterminate'];
}
