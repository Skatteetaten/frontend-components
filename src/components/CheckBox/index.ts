export { CheckBox as default } from './CheckBox';
export * from './CheckBox';
