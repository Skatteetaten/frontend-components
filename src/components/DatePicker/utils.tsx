export const DEFAULTFORMATDATE = (date: Date | null | undefined): string => {
  return date
    ? Intl.DateTimeFormat('no', {
        year: 'numeric',
        month: '2-digit',
        day: 'numeric',
      }).format(date)
    : '';
};

export const DEFAULTPARSEDATEFROMSTRING = (date: string): Date | null => {
  try {
    if (date && date.length === 10) {
      const day: number = parseInt(date.substring(0, 2));
      const month: number = parseInt(date.substring(3, 5));
      const year: number = parseInt(date.substring(6, 10));
      const formattedDate: Date = new Date(Date.UTC(year, month - 1, day));
      return formattedDate;
    }
  } catch (error) {
    console.log(error);
  }
  return null;
};

export const monthsForLocale = (
  localeName = 'no',
  monthFormat: 'long' | 'short' = 'long'
) => {
  const format = new Intl.DateTimeFormat(localeName, { month: monthFormat })
    .format;
  const months: string[] = [];
  for (let month = 0; month < 12; month++) {
    months.push(format(new Date(Date.UTC(2020, month))));
  }
  return months;
};

export const weekdaysForLocale = (
  localeName = 'no',
  dayFormat: 'long' | 'short' = 'long'
) => {
  const format = new Intl.DateTimeFormat(localeName, { weekday: dayFormat })
    .format;
  const weekdays: string[] = [];
  for (let day = 0; day < 7; day++) {
    weekdays.push(format(new Date(Date.UTC(2020, 7, day + 2)))); // add 2 days to let the week start on Sunday
  }
  return weekdays;
};
