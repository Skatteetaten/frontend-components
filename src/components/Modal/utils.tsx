import { useEffect } from 'react';

export const createModalDomPlacement = (
  setFlag: (a: boolean) => void,
  shadowRootNode: Document | ShadowRoot | undefined
) => {
  if (!shadowRootNode) {
    setFlag(true);
    return;
  }

  const styleInjectorWrapperId = 'style-injector';
  const modalWrapperId = 'modal-wrapper';

  if (shadowRootNode.getElementById(modalWrapperId)) {
    setFlag(true);
  } else {
    const modalWrapper = document.createElement('div');
    modalWrapper.id = modalWrapperId;
    const styleInjectorWrapper = shadowRootNode.getElementById(
      styleInjectorWrapperId
    );
    const parent = styleInjectorWrapper ? styleInjectorWrapper : shadowRootNode;
    parent.appendChild(modalWrapper);
    setFlag(true);
  }
};

export const useEscOnPress = (
  ref: HTMLDivElement | null,
  onClose?: (ref: HTMLDivElement) => void
): void => {
  useEffect(() => {
    const onKeyDown = (event: KeyboardEvent): void => {
      if (ref == null) {
        return;
      }
      if (event.code === 'Escape' && onClose) {
        onClose(ref);
      }
    };

    if (ref) {
      window.addEventListener('keydown', onKeyDown as EventListener, false);
    }
    return function cleanup(): void {
      if (window) {
        window.removeEventListener('keydown', onKeyDown as EventListener);
      }
    };
  }, [ref, onClose]);
};
