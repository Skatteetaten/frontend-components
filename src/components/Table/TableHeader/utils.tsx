import { TFunction } from 'react-i18next';

export const getIconNameForTheadCell = (
  isSorted: boolean,
  isSortedAscending: boolean
) => {
  if (isSorted) {
    return isSortedAscending ? 'ArrowDown' : 'ArrowUp';
  }

  return 'ArrowUpDown';
};

export const getAriaLabelForTheadCell = (
  name: string | JSX.Element,
  fieldName: string,
  isSorted: boolean,
  isSortedAscending: boolean,
  t: TFunction<Array<string>>
) => {
  let ariaLabel = fieldName;
  if (typeof name === 'string') {
    ariaLabel = isSorted
      ? isSortedAscending
        ? name.concat(' ', t('table.sorted_ascending'))
        : name.concat(' ', t('table.sorted_descending'))
      : name.concat(' ', t('table.sortable'));
  }

  return ariaLabel;
};
