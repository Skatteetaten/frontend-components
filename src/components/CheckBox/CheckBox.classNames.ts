import { mergeStyles } from '@uifabric/merge-styles';
import { getTheme } from '@uifabric/styling';
import { getFocusStyle } from '../utils/getFocusStyle';
import { FontWeights, FontSizes } from '../utils/fonts';
// @ts-ignore TODO
export var getClassNames = function getClassNames(props) {
  const theme = getTheme();
  const inset = -4;
  const radius = '0';
  // @ts-ignore TODO
  return mergeStyles([
    getFocusStyle(theme, inset, 'relative', radius),
    {
      displayName: 'SkeAvkrysningsboks',
      selectors: {
        '&.ms-CheckBox': {
          padding: '4px'
        },
        '&.is-enabled .ms-CheckBox-checkbox': {
          borderRadius: '0px',
          content: '',
          display: 'inline-block',
          border: '2px solid',
          cursor: 'pointer',
          fontSize: FontSizes.medium
        },
        'span.ms-CheckBox-text': {
          fontSize: FontSizes.medium
        },
        '&.is-checked .ms-CheckBox-checkbox': {
          border: 'none',
          fontSize: FontSizes.medium
        },
        'i.ms-CheckBox-checkmark': {
          paddingLeft: '1.5px',
          paddingTop: '1.5px',
          fontSize: FontSizes.medium
        },
        '&:hover .ms-CheckBox-checkmark': {
          opacity: 0
        },
        '&.is-checked:hover .ms-CheckBox-checkmark': {
          opacity: 1
        },
        '& i': {
          fontWeight: FontWeights.bold
        },
        '.ms-CheckBox-label:hover i': {
          opacity: 0
        }
      }
    }
  ]);
};
