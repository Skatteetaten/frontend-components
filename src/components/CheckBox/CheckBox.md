** Avkrysningsboks brukes for eksempel i skjemaer til å bekrefte et eller flere valg. **

```js
import CheckBox from '@skatteetaten/frontend-components/CheckBox';

<div>
  <fieldset style={{ border: 'none' }}>
    <legend style={{ fontSize: '14px' }}>Velg aktuelle lover</legend>
    <CheckBox boxSide={'start'} label="Skatteloven" />
    <CheckBox boxSide={'start'} label="Skattebetalingsloven" />
    <CheckBox boxSide={'start'} label="Skatteforvaltningsloven" />
  </fieldset>
</div>;
```

```js noeditor
import Accordion from '@skatteetaten/frontend-components/Accordion';
import AccordionItem from '@skatteetaten/frontend-components/Accordion/AccordionItem';

<Accordion>
  <AccordionItem
    toggleContent
    isOpen
    toggleButtonText={'Bruk'}
    stepId={'step-1-1'}
  >
    <p>
      Avhukingsbokser brukes når det er liste med valg og brukeren kan velge et
      vilkårlig antall av dem, inkludert null, ett eller flere. En avhukingsboks
      er med andre ord uavhengig av de andre i listen. Og huker brukeren av et
      valg, tar det ikke bort avhukingen av de andre.
    </p>

    <p>(Hvis brukeren kun skal velge ett valg, bruk RadioButtonGroup.)</p>
  </AccordionItem>
  <AccordionItem
    toggleContent
    toggleButtonText={'Universell utforming'}
    stepId={'step-1-2'}
  >
    <ul>
      <li>
        Sjekk at tilstand på avhuking (valgt/ikke valgt) leses opp av
        skjermleser
      </li>
      <li>
        Bruk <b>fieldset</b> og <b>legend</b>-elementer for å gi
        kontekstinformasjon til skjermleser (hvilke elementer hører sammen, og
        hva er overordnet tittel eller emne).
      </li>
    </ul>
  </AccordionItem>
  <AccordionItem
    toggleContent
    toggleButtonText={'Tekniske detaljer'}
    stepId={'step-1-3'}
  >
    <p>
      <a
        href="https://developer.microsoft.com/en-us/fabric#/components/checkbox#Implementation"
        target="_blank"
      >
        Se flere tilgjengelige props i Fabric dokumentasjonen
      </a>
    </p>
  </AccordionItem>
</Accordion>;
```
