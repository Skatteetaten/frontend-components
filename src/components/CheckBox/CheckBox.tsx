import classnames from 'classnames';
import {
  Checkbox as FabricCheckBox,
  ICheckboxProps
} from 'office-ui-fabric-react/lib-commonjs/Checkbox';
import * as React from 'react';
import { getClassNames } from './CheckBox.classNames';

interface CheckBoxProps extends ICheckboxProps {
  /** Angir teknisk navn p√• elementet */
  name?: string;
}
/**
 * @visibleName CheckBox (Avhukingsboks)
 */
export default class CheckBox extends React.PureComponent<CheckBoxProps, {}> {
  static defaultProps = {
    boxSide: 'end'
  };
  render() {
    const { checked, defaultChecked, className, ...props } = this.props;
    const checkedProps = defaultChecked ? { defaultChecked } : { checked };
    return (
      <FabricCheckBox
        {...props}
        className={classnames(getClassNames(this.props), className)}
        role="checkbox"
        {...checkedProps}
      />
    );
  }
}
