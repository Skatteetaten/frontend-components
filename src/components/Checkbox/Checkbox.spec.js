import React from 'react';
import { matches } from './../utils/test-utils';
import { shallow } from 'enzyme';
import CheckBox from './CheckBox';

function oppsettShallow(props) {
  return shallow(<CheckBox {...props} />);
}

describe('Checkbox komponent', () => {
  it('matcher snapshot', () => {
    matches(<CheckBox>Sjekkboks</CheckBox>);
  });

  it('rendrer Checkbox med default props', () => {
    const wrapper = shallow(<CheckBox />);
    expect(wrapper.find('StyledCheckboxBase').prop('boxSide')).toEqual('end');
  });

  it('setter Checkbox med riktige props', () => {
    const wrapper = oppsettShallow({
      checked: true,
      disabled: true,
      boxSide: 'start',
      label: 'checkbox-label',
      name: 'checkbox-name',
      id: 'checkbox-id',
      className: 'checkbox-classname'
    });

    const checkbox = wrapper.find('StyledCheckboxBase');

    expect(checkbox.prop('checked')).toEqual(true);
    expect(checkbox.prop('disabled')).toEqual(true);
    expect(checkbox.prop('boxSide')).toEqual('start');
    expect(checkbox.prop('label')).toEqual('checkbox-label');
    expect(checkbox.prop('name')).toEqual('checkbox-name');
    expect(wrapper.prop('id')).toEqual('checkbox-id');
    expect(checkbox.prop('className')).toContain('checkbox-classname');
  });
});
