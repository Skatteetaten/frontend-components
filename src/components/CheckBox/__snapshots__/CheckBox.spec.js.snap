// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Checkbox komponent matcher snapshot 1`] = `
<div
  className="ms-Checkbox reversed is-enabled SkeAvkrysningsboks-37"
>
  <input
    aria-checked="false"
    className="input-38"
    id="checkbox-0"
    onBlur={[Function]}
    onChange={[Function]}
    onFocus={[Function]}
    type="checkbox"
  />
  <label
    className="ms-Checkbox-label label-39"
    htmlFor="checkbox-0"
  >
    <div
      className="ms-Checkbox-checkbox checkbox-40"
    >
      <i
        aria-hidden={true}
        className="ms-Checkbox-checkmark checkmark-44"
        data-icon-name="CheckMark"
      >
        ï„¬
      </i>
    </div>
  </label>
</div>
`;
